src/gtk/toolbar.cpp

```diff
@@ -644,6 +644,14 @@ bool wxToolBar::DoInsertTool(size_t pos, wxToolBarToolBase *toolBase)
 #else
             tool->m_item = GTK_TOOL_ITEM(gtk_widget_get_parent(gtk_widget_get_parent(control->m_widget)));
 #endif
+            // The widget size is not controlled by wx, so at least make sure
+            // that its minimal size is respected by GTK.
+            const wxSize minSize = control->GetBestSize();
+            if ( minSize != wxDefaultSize )
```

Comment:
Minor, but the check isn't needed. -1 means the same thing on both sides, and nothing will happen if the values aren't changing.